### DISCUSSION TITLE
Branch Protection

### DISCUSSION THREAD

**Discussion by Woon-2** [MyeongWoonJang/cpp-study#36] [Announcements]
이 프로젝트는 다음의 `Branch Protection Rule`을 사용합니다.

- `main` 과 `-protected`로 끝나는 이름의 `Branch`는 임의로 `Merge`할 수 없고,   
  `Pull Request`를 생성해야 합니다.

  - 최소 1개 이상의 승인이 필요합니다.
  - 승인을 얻은 `Pull Request`에 새로이 `Commit`이 추가된다면, 재승인 받아야 합니다.
  - `Code Owners`(추후 추가 예정인 CODEOWNER 파일의 명세를 따릅니다.)의 `Review`가 완료되어야   
  `Pull Request`를 `Merge`할 수 있습니다.

- `Pull Request`의 `Conversation` 항목에서 모든 `Conversation`에 대해 `Resolve Conversation` 버튼을 눌러야   
  `Pull Request`를 `Merge`할 수 있습니다.

<div align="center"><image src="https://user-images.githubusercontent.com/73771162/194463392-648e9e76-6e6c-4b84-8136-39a91c941a45.PNG"></div>

<div align="center"><image src="https://user-images.githubusercontent.com/73771162/194463395-e519508d-3db1-4550-be59-a589872e87f8.PNG"></div>

**Comment by Lcat12345** [comment]
새로운 규칙이 생긴거군요
